### PR TITLE
feat: add new textspliter option: lenfunc

### PR DESCRIPTION
**Pull request by whyiug**
I want to add a new parameter to the textspliter struct,**lenfunc**, to represent a custom length function. In this way, sentences can be splited according to lenfuc, for example,  ` utf8.RuneCountInString` can be passed to split Chinese characters.
Also aligned with the python code https://github.com/langchain-ai/langchain/blob/00a09e1b7117f3bde14a44748510fcccc95f9de5/libs/langchain/langchain/text_splitter.py.
For me, it's a must-have feature.

### CODE REVIEW

**Comment by whyiug** [comment]
Also satisfy issue #231

**Comment by whyiug** [comment]
ping @tmc

**Comment by tmc** [comment]
Hmm, it might be best to just default to RuneCount/RuneCountInString to treat content more naturally, thoughts?

I think I prefer this over erroneously using len() on strings.

**Comment by tmc** [comment]
This is better, but I can't help but think we shouldn't even make this configurable and just use utf8.RuneCountInString -- thoughts?

File: textsplitter/options.go [hunk 1]
```diff
@@ -20,6 +23,7 @@ func DefaultOptions() Options {
 		ChunkSize:    _defaultTokenChunkSize,
 		ChunkOverlap: _defaultTokenChunkOverlap,
 		Separators:   []string{"\n\n", "\n", " ", ""},
+		LenFunc:      defaultLenFunc,
```
Comment (tmc) [inline textsplitter/options.go:26]: this could just be `utf8.RuneCountInString` instead of being wrapped with defaultLenFunc, I believe.

**Comment by whyiug** [comment]
Hmm, I don't think so. 
Some scenarios require custom length functions. 
Let's say I want to have the same number of words, not characters  in each chunk.
I can simply set the LenFunc:
```go
func countWords(s string) int {
    words := strings.Fields(s)
    return len(words)
}
```
Or the number of tokens divided by other word dividers.

**Review by tmc — APPROVED** [review]
LGTM

**Pull request merged**